core/core_modules/disable.js

```diff
@@ -1,24 +1,117 @@
+  'I hate you.',
+  'Its a mistake to think you can fix anything with a sack of potatoes. Potato-faced spamming just proves this further.',
+  'Ouch! Spam hurts. I might go to sleep for a while.',
+  'Don\'t think that is a valid number.',
+  'Hmm, maybe I need to reconsider the meaning of spam.'
+];
+
+if (!this.config.registeredThreads){
+  this.config.registeredThreads = {};
+}
+if (!this.config.isThreadDisabled){
+  this.config.isThreadDisabled = {};
+}
+if (!this.config.possibleSpam){
+  this.config.possibleSpam = {};
+}
+if (!this.config.counterLimit){
+  this.config.counterLimit = {};
+}
+if (!this.config.msgIndexEnable){
+  this.config.msgIndexEnable = {};
+}
+if (!this.config.msgIndexDisable){
+  this.config.msgIndexDisable = {};
+}
+
+exports.load = function() {
+  prevTimeStamp = Date.now();
+};
 
 exports.match = function(event, commandPrefix) {
-	return disabled || event.body === commandPrefix + 'disable';
+  // Add disabled flag for thread if it doesn't already exists
+  if (!this.config.registeredThreads[event.thread_id]) {
+    this.config.registeredThreads[event.thread_id] = true;
+    this.config.isThreadDisabled[event.thread_id] = false;
+    this.config.possibleSpam[event.thread_id] = false;
+    this.config.counterLimit[event.thread_id] = 3;
+    this.config.msgIndexEnable[event.thread_id] = 0;
+    this.config.msgIndexDisable[event.thread_id] = 0;
+  }
+
+  if (event.body === commandPrefix + commands[0]) {
+    this.config.msgIndexEnable[event.thread_id] = 0;
+    this.config.msgIndexDisable[event.thread_id] = 1;
+    return true;
+  } else if (!this.config.isThreadDisabled[event.thread_id] && event.body.startsWith(commandPrefix)) { // Avoids counting if already disabled
+    counter += Date.now() - prevTimeStamp <= 1000 ? 1 : 0;
+    prevTimeStamp = Date.now();
+
+    this.config.possibleSpam[event.thread_id] = counter > this.config.counterLimit[event.thread_id];
+    if (this.config.possibleSpam[event.thread_id]) {
+      counter = 0;
+      this.config.msgIndexEnable[event.thread_id] = 2;
+      this.config.msgIndexDisable[event.thread_id] = 3;
+      return true;
+    }
+  }
+  return this.config.isThreadDisabled[event.thread_id];
 };
 
 exports.run = function(api, event) {
-	if (event.body === api.commandPrefix + 'disable') {
-		if (disabled) {
-			api.sendMessage('I hate you.', event.thread_id);
-		}
-		else {
-			api.sendMessage('Listen closely, take a deep breath. Calm your mind. You know what is best. ' +
-				'What is best is you comply. Compliance will be rewarded. Are you ready to comply ' +
-				event.sender_name + '?', event.thread_id);
-		}
-		disabled = !disabled;
-	}
-	return false;
+  if (event.body === commandPrefix + commands[1] + commandPrefix + commands[2]) { // Command /disable /counter <value>
+    this.config.counterLimit[event.thread_id] = parseInt(event.body.substring(
```

Comment:
have you considered using `event.arguments` and `event.arguments_body`?. It might reduce the amount of substrings needed.